tests/e2e/multi_vc_utils.go

```diff
@@ -0,0 +1,433 @@
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return
+			} else {
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}
+	}
+}
+
+/*
+verifyVolumeMetadataInCNSForMultiVC verifies container volume metadata is matching the
+one is CNS cache on a multivc environment.
+*/
+func verifyVolumeMetadataInCNSForMultiVC(vs *multiVCvSphere, volumeID string,
+	PersistentVolumeClaimName string, PersistentVolumeName string,
+	PodName string, Labels ...vim25types.KeyValue) error {
+	queryResult, err := vs.queryCNSVolumeWithResultInMultiVC(volumeID)
+	if err != nil {
+		return err
+	}
+	gomega.Expect(queryResult.Volumes).ShouldNot(gomega.BeEmpty())
+	if len(queryResult.Volumes) != 1 || queryResult.Volumes[0].VolumeId.Id != volumeID {
+		return fmt.Errorf("failed to query cns volume %s", volumeID)
+	}
+	for _, metadata := range queryResult.Volumes[0].Metadata.EntityMetadata {
+		kubernetesMetadata := metadata.(*cnstypes.CnsKubernetesEntityMetadata)
+		if kubernetesMetadata.EntityType == "POD" && kubernetesMetadata.EntityName != PodName {
+			return fmt.Errorf("entity Pod with name %s not found for volume %s", PodName, volumeID)
+		} else if kubernetesMetadata.EntityType == "PERSISTENT_VOLUME" &&
+			kubernetesMetadata.EntityName != PersistentVolumeName {
+			return fmt.Errorf("entity PV with name %s not found for volume %s", PersistentVolumeName, volumeID)
+		} else if kubernetesMetadata.EntityType == "PERSISTENT_VOLUME_CLAIM" &&
+			kubernetesMetadata.EntityName != PersistentVolumeClaimName {
+			return fmt.Errorf("entity PVC with name %s not found for volume %s", PersistentVolumeClaimName, volumeID)
+		}
+	}
+	labelMap := make(map[string]string)
+	for _, e := range queryResult.Volumes[0].Metadata.EntityMetadata {
+		if e == nil {
+			continue
+		}
+		if e.GetCnsEntityMetadata().Labels == nil {
+			continue
+		}
+		for _, al := range e.GetCnsEntityMetadata().Labels {
+			labelMap[al.Key] = al.Value
+		}
+		for _, el := range Labels {
+			if val, ok := labelMap[el.Key]; ok {
+				gomega.Expect(el.Value == val).To(gomega.BeTrue(),
+					fmt.Sprintf("Actual label Value of the statically provisioned PV is %s but expected is %s",
+						val, el.Value))
+			} else {
+				return fmt.Errorf("label(%s:%s) is expected in the provisioned PV but its not found", el.Key, el.Value)
+			}
+		}
+	}
+	ginkgo.By(fmt.Sprintf("successfully verified metadata of the volume %q", volumeID))
+	return nil
+}
+
+// govc login cmd for multivc setups
+func govcLoginCmdForMultiVC() string {
+	configUser := strings.Split(multiVCe2eVSphere.multivcConfig.Global.User, ",")
+	configPwd := strings.Split(multiVCe2eVSphere.multivcConfig.Global.Password, ",")
+	configvCenterHostname := strings.Split(multiVCe2eVSphere.multivcConfig.Global.VCenterHostname, ",")
+	configvCenterPort := strings.Split(multiVCe2eVSphere.multivcConfig.Global.VCenterPort, ",")
+
+	loginCmd := "export GOVC_INSECURE=1;"
+	loginCmd += fmt.Sprintf("export GOVC_URL='https://%s:%s@%s:%s';",
```

Comment:
This command is connecting to VC1 only, What if we want to connect to other 2 VC's

Reply:
Yeah, you will see the fix for this in the next PR for preferential and config secret.
Since these test cases do not require other VC connections, I kept it like this
But in the next PR which I will be sending for review, you will find these utils get controlled through iterator variable